test/functional/measures_controller_test.rb

```
@@ -760,15 +760,20 @@ class << update_measure_file
   end
 
   test 'update measurement period' do
-    measure = CQM::Measure.first
+    load_measure_fixtures_from_folder(File.join('measures', 'CMS32v7'), @user)
+    measure = CQM::Measure.where({cms_id: "CMS32v7"}).first
     measure_id = measure.id
     assert_equal '2012', measure.measure_period['low']['value'].slice(0,4)
     post :measurement_period, {
       year: '1984',
-      id: measure.id.to_s
+      id: measure.id.to_s,
+      measurement_period_shift_dates: "true"
     }
     measure = CQM::Measure.where(id: measure_id).first
     assert_equal '1984', measure.measure_period['low']['value'].slice(0,4)
+    patient = CQM::Patient.by_user(@user).first
+    assert_equal 1984, patient.qdmPatient.dataElements.first.authorDatetime.year
+    assert_equal 1984, patient.qdmPatient.dataElements.first.relevantPeriod.low.year
```

Should add a test where the data element had a year other than 2012 originally and check that it has shifted appropriately by the correct number of years

One edge case that is unlikely but worth considering is if the shift in dates would produce an invalid CQL Year,
for example if a data element has year 0001 but then the MP is changed from 2012 to 2011.  We might want to throw an error in that case